src/jvmMain/kotlin/io/github/vyfor/kpresence/ipc/Connection.kt

```diff
@@ -68,15 +68,19 @@ actual class Connection {
     override fun read(): Message? {
       pipe?.let { stream ->
         try {
-          val opcode = stream.readInt(0).reverseBytes()
-          val length = stream.readInt(4).reverseBytes()
-          val buffer = ByteBuffer.allocate(length)
-          
-          stream.read(buffer, 8).get()
-          return Message(
-            opcode,
-            buffer.array()
-          )
+          try {
+            val opcode = stream.readInt(0).reverseBytes()
+            val length = stream.readInt(4).reverseBytes()
+            val buffer = ByteBuffer.allocate(length)
+
+            stream.read(buffer, 8).get()
+            return Message(
+              opcode,
+              buffer.array()
+            )
+          } catch (e: ExecutionException) {
+            throw (e.cause ?: throw PipeReadException(e.message.orEmpty()))
+          }
```

Comment:
```suggestion
        } catch (e: ExecutionException) {
          throw (e.cause ?: throw PipeReadException(e.message.orEmpty()))
        }
```

Reply:
Making that change won't work. The reason for the nested try-catch is to rethrow the cause of `ExecutionException` for the "parent" try-catch to handle it.